kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaEventListener.java

```diff
@@ -40,14 +52,17 @@ public Builder<K, V> registerCallback(BiConsumer<? super K, ? super V> callbackF
 
     public Builder<K, V> withExecutorService(ExecutorService executorService) {
       this.executorService = executorService;
+      this.cleanupExecutor = false;
```

Comment:
The way we typically would use this, the executor - even if custom - would be disposed of with the listener. If we want to support either way, add this as a flag to withExecutorService, but if we're just going to default I'd default to true.